calc.js

```diff
@@ -1,3 +1,85 @@
-// Build a simple webpage that has an interactive calculator using JavaScript
+let input = document.getElementById("Input");
+let HISTORY = "";
+const clear = document.getElementById("Clear");
+const allClear = document.getElementById("allClear");
+const equal = document.getElementById("Equal");
+const calcButtons = document.querySelectorAll(".calcButton");
 
-// You will want some kind of html element to display the calculations (maybe a span) and a few buttons to press to input numbers and operators
\ No newline at end of file
+clear.addEventListener("click", clearInput);
+allClear.addEventListener("click", clearAll);
+equal.addEventListener("click", doEquation);
+calcButtons.forEach(addEventListenerToCalcButtons);
```

Comment:
function is called before it's created, @oakeef is this fine? I found conflicting info when I looked it up but looks like it might just be something that was added to JS more recently

Reply:
it's because JS is an interpreted language and not a compiled one. The browser reads the whole JS file before running it I believe

Reply:
Makes sense. Thank you!